crates/electrum/tests/test_electrum.rs

```diff
@@ -272,32 +292,91 @@ fn scan_detects_confirmed_tx() -> anyhow::Result<()> {
     // Mine some blocks.
     env.mine_blocks(101, Some(addr_to_mine))?;
 
-    // Create transaction that is tracked by our receiver.
+    // Broadcast transaction to mempool.
     env.send(&addr_to_track, SEND_AMOUNT)?;
+    env.wait_until_electrum_sees_block()?;
+
+    electrum_sync(
+        &client,
+        &[spk_to_track.clone()],
+        5,
+        &mut recv_chain,
+        &mut recv_graph,
+    )?;
+
+    // Check if balance is zero when transaction exists only in mempool.
+    assert_eq!(
+        get_balance(&recv_chain, &recv_graph)?,
+        Balance {
+            confirmed: Amount::from_sat(0),
+            ..Balance::default()
+        },
+        "initial balance must be correct",
+    );
```

Comment:
Since we have a tx in mempool, shouldn't we have an unconfirmed balance?

If the intention is to sync BEFORE seeing the tx in mempool, it should be made more obvious in code.